evaluator.plandefinition/src/test/java/org/opencds/cqf/cql/evaluator/plandefinition/r4/MockFhirDal.java

```diff
@@ -0,0 +1,187 @@
+package org.opencds.cqf.cql.evaluator.plandefinition.r4;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.instance.model.api.IIdType;
+import org.hl7.fhir.r4.model.ActivityDefinition;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Encounter;
+import org.hl7.fhir.r4.model.PlanDefinition;
+import org.hl7.fhir.r4.model.ValueSet;
+import org.opencds.cqf.cql.evaluator.fhir.dal.FhirDal;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.context.FhirVersionEnum;
+import ca.uhn.fhir.parser.IParser;
+
+public class MockFhirDal implements FhirDal {
```

Comment:
This class should be split into two parts:
1. Bundle based FHIR dal
2. Loading a resource as a Bundle